File: 21-merge-two-sorted-lists/merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode head = new ListNode();
        ListNode dummy = head;
        while(list1 != null && list2 != null) {
            if(list1.val <= list2.val) {
                ListNode v = new ListNode(list1.val);
                dummy.next = v;
                list1 = list1.next;
            } else {
                 ListNode v = new ListNode(list2.val);
                 dummy.next = v;
                 list2 = list2.next;
            }
            dummy = dummy.next;
        }
        dummy.next = (list1 != null) ? list1 : list2;
        return head.next;
    }
}

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode(-1)
        dummy = head
        while list1 is not None and list2 is not None:
            if list1.val <= list2.val:
                dummy.next = list1
                list1 = list1.next
            else:
                dummy.next = list2
                list2 = list2.next
            dummy = dummy.next
        while list1 is not None:
            dummy.next = list1
            list1 = list1.next
            dummy = dummy.next
        while list2 is not None:
            dummy.next = list2
            list2 = list2.next
            dummy = dummy.next
        return head.next